fl4health/privacy_mechanisms/dpSkellam.py

```diff
@@ -0,0 +1,58 @@
+import torch
+
+
+def SkellamMechanism(query_vector: list[int], skellam_variance: float) -> list[int]:
+    """
+     An additive mechanism that adds to the given query vector a discrete noise vector
+     sampled from the centered Skellam distribution of given variance.
+
+
+     Args:
+         query_vector (list[int]): Discritized vector to which noise will be added.
+         skellam_variance (float): Variance associated with each component of the d-dimensional Skellam distribution.
+
+     Raises:
+         Exception: Empty query
+         Exception: Un-discretized query.
+
+     Returns:
+         list[int]: Perturbed vector.
+
+    Notes
+         1) A Skellam random variable can be obtained as the difference of two independent Poisson random variables.
+         2) A Skellam random vector has as components Skellam random variables.
+         3) Computations are carried throughout on torch.int (32-bit signed integer)
+
+     Reference
+         The Skellam Mechanism for Differentially Private Federated Learning
+         https://proceedings.neurips.cc/paper/2021/file/285baacbdf8fda1de94b19282acd23e2-Paper.pdf
+    """
+
+    dim = len(query_vector)
+
+    if dim == 0:
+        raise Exception("query_vector cannot be empty")
+
+    if type(query_vector[0]) != int:
+        raise Exception("Did you forget to discretize the gradient vector? query_vector must be a list of integers")
+
+    # mean of Possion variables used to construct a
+    # centered Skellam variable S with Var(S) = skellam_variance
+    mean = skellam_variance / 2
```

Comment:
Maybe name this something like `poisson_mean` to make it explicit that this is the poisson expectation we're going to use.

It also might be worth noting that Var(S) = \mu_1 + \mu_2 when S ~ Poisson(mu_1) - Poisson(mu_2)?